ops/model.py

```diff
@@ -1047,21 +1047,25 @@ def stop(self, *service_names: str):
         """Stop given service(s) by name."""
         self._pebble.stop_services(service_names)
 
-    def add_layer(self, layer: typing.Union[str, typing.Dict, 'pebble.Layer']):
-        """Dynamically add a setup layer.
+    def add_layer(self, label: str, layer: typing.Union[str, typing.Dict, 'pebble.Layer'],
+                  combine: bool = False):
+        """Dynamically add a new layer onto the Pebble configuration layers.
 
         Args:
-            layer: A YAML string, setup layer dict, or pebble.Layer object
-                containing the Pebble layer to add.
+            label: Label for new layer (and label of layer to merge with if
+                combining).
+            layer: A YAML string, configuration layer dict, or pebble.Layer
+                object containing the Pebble layer to add.
+            combine: If combine is False (the default), append the new layer
+                as the top layer with the given label. If combine is True,
+                combine the new layer with the layer that has the given label;
+                combining is done according to Pebble's override rules.
```

Comment:
That last sentence makes it feel like Pebble will decide what to do based on its rules, but it will actually combine layers based on what the layer itself asked to do for with particular service.

How about: "(...) If combine is True and the label already exists, the two layers are combined in a single one considering the layer override rules. If the layer doesn't exist, it is added as usual."

Reply:
Done.